src/shared/sam/localLambdaRunner.ts

```diff
@@ -689,7 +703,8 @@ export async function makeJsonFiles(config: SamLaunchRequestArgs): Promise<void>
     // env-vars.json (NB: effectively ignored for the `target=code` case).
     const configEnv = config.lambda?.environmentVariables ?? {}
     if (Object.keys(configEnv).length !== 0) {
-        const env = JSON.stringify(getEnvironmentVariables(makeResourceName(config), configEnv))
+        const resourceName = await makeResourceName(config)
+        const env = JSON.stringify(getEnvironmentVariables(resourceName, configEnv))
```

Comment:
This is the one place that the resource name is needed for `target=api` case. Doesn't seem avoidable?

Reply:
> SAM CLI should be doing (?)

Definitely. This code was originally a workaround for the fact that we generate a temporary, partial template.yaml instead of using the identical user-defined one...

...however I forgot that we changed the `target=template` case to use the "identical user-defined template.yaml".  https://github.com/aws/aws-toolkit-vscode/pull/2061#discussion_r714407617
This is great! 

Need to update the docstring for `makeJsonFiles()`, I think it's out of date. And then I think `getResourceByApiPath()` is not needed!

Reply:
oh wait, `env-vars.json` is for `lambda.environmentVariables`. I suppose we could just not support `lambda.environmentVariables` for `target=api`.

Reply:
Ah right the variables could come from the launch config. Could we just not merge the variables? If no resource is known, then it's just pass-through to SAM CLI. I would expect SAM CLI to then merge the variables when invoking the lambda, but maybe they don't.

Reply:
>  I believe the environment variables file is treated as 'global' if no resource name is specified.

That seems perfect. We don't have per-resource concept in launch config so there is no reason to do that in `env-vars.json`.